### ISSUE TITLE
`transpose` crashes on mps device

### ISSUE THREAD

**Issue by ducha-aiki** [pytorch/pytorch#77958]
### 🐛 Describe the bug

Pytorch crashed with segfault when using transpose on `mps` device
```python3
import torch
a=torch.ones(1,2,2, device=torch.device('mps'))
print (a)
print (a.transpose(2,1))
```

Expected: not crash

Outcome:

```
tensor([[[1., 1.],
         [1., 1.]]], device='mps:0')
-:12:10: error: 'mps.get_coordinates' op invalid axis: -2.
-:12:10: note: see current operation: %9 = "mps.get_coordinates"(%1, %8) : (tensor<1xi32>, tensor<i32>) -> tensor<2xi32>
segmentation fault  
```

### Versions

```
Collecting environment information...
PyTorch version: 1.12.0.dev20220520
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.3 (arm64)
GCC version: Could not collect
Clang version: 13.0.1
CMake version: version 3.22.3
Libc version: N/A

Python version: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:10:52)  [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.3-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] pytest-mypy==0.9.0
[pip3] torch==1.12.0.dev20220520
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.10.0
[conda] numpy                     1.21.2                   pypi_0    pypi
[conda] torch                     1.12.0.dev20220520          pypi_0    pypi
[conda] torchaudio                0.11.0                   pypi_0    pypi
[conda] torchvision               0.10.0                   pypi_0    pypi
```

**Comment by kulinseth** [comment]
Can you please check @ducha-aiki , the issue is fixed in : torch-1.13.0.dev20220524

**Comment by MishaRuko** [comment]
Transposing tensors with 3 or more dimensions still results in a crash with torch-1.13.0.dev20220524 
```python
import torch
x = torch.rand((10, 5, 4), device=torch.device("mps"))
x.transpose(0, 1)
```
